observability/test/run_o11y_tests_local.py

```diff
@@ -0,0 +1,96 @@
+import docker # type: ignore
+import logging
+import os
+import subprocess
+import sys
+from test_utils import (
+    ObservabilityTestCase
+)
+
+SPONGE_LOGS_DIR = '/tmp/observability_test_log' # a directory in your local environment for logs
+DOCKER_IMAGE_NAME = 'gcr.io/microsvcs-testing/grpc-observability/testing/integration-%s:latest'
+
+argp = argparse.ArgumentParser(description='Run Observability integration tests in local env')
+argp.add_argument('--server_lang', required=True, type=str,
+                  help='Server language: java | go | cpp')
+argp.add_argument('--client_lang', required=True, type=str,
+                  help='Client language: java | go | cpp')
+argp.add_argument('--test_case', required=True, type=str,
+                  help='Test case to run: check test_utils.py')
+argp.add_argument('--docker_image_go', default=DOCKER_IMAGE_NAME % 'go', type=str,
+                  help='docker image tag for Go interop client/server')
+argp.add_argument('--docker_image_java', default=DOCKER_IMAGE_NAME % 'java', type=str,
+                  help='docker image tag for Java interop client/server')
+argp.add_argument('--docker_image_cpp', default=DOCKER_IMAGE_NAME % 'cpp', type=str,
+                  help='docker image tag for C++ interop client/server')
+args = argp.parse_args()
+
+docker_client = docker.from_env()
+
+logger = logging.getLogger(__name__)
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(fmt='%(asctime)s: %(levelname)-8s %(message)s')
+console_handler.setFormatter(formatter)
+logger.handlers = []
+logger.addHandler(console_handler)
+logger.setLevel(logging.DEBUG)
+
+def prepare_docker_image(lang):
+    args_dict = vars(args)
+    image_name = args_dict['docker_image_%s' % lang]
+    images = docker_client.images.list(name=image_name)
+    if len(images) == 0:
+        logger.warning("Docker image '%s' does not exist locally. Trying to docker pull ..." % image_name)
+        try:
+            if not docker_client.images.pull(repository=image_name):
+                raise Exception('No image found')
+        except:
+            logger.error("Still could not find docker image '%s'. Exiting...'" % image_name)
+            sys.exit(1)
+    logger.info("Using local docker image '%s'" % image_name)
+    os.environ['OBSERVABILITY_TEST_IMAGE_%s' % lang.upper()] = image_name
+
+def main():
+    os.makedirs(SPONGE_LOGS_DIR, exist_ok=True)
+    prepare_docker_image(args.server_lang)
+    prepare_docker_image(args.client_lang)
+    if args.test_case == 'all':
+        for test_case in ObservabilityTestCase:
+            run_test_case(test_case)
+    else:
+        run_test_case(args.test_case)
+
+def run_test_case(test_case):
+    os.environ['RESOURCE_TYPE_ASSERTION_OVERRIDE'] = 'global'
+    os.environ['CUSTOM_DOCKER_RUN_AUTH'] = \
+        '-v %s/.config/gcloud:/root/.config/gcloud' % os.environ.get('HOME')
+    os.environ['KOKORO_ARTIFACTS_DIR'] = SPONGE_LOGS_DIR
+    proc = subprocess.Popen(
```

Comment:
Nit: Instead of subprocessing out, you could create a top-level function in `run_o11y_tests.py` that takes individual arguments rather than flags, then in this file, run `import run_o11y_tests` and call `run_o11y_tests.main(...)`. Not strictly necessary, but whenever I subprocess, I usually end up with a debugging effort at some point in the future.

Reply:
Noted. May change in the future.